### PR TITLE
PlainDecoder take on boolean values

### PR DESCRIPTION
**Pull request by eddyxu**
Support run `take()` random access on boolean columns

### CODE REVIEW

File: rust/src/encodings/plain.rs [hunk 1]
```diff
@@ -178,7 +186,7 @@ impl<'a> PlainDecoder<'a> {
             )));
         }
         let start_offset = make_byte_offset(self.data_type, start)?;
-        let end_offset = make_byte_offset(self.data_type, end)?;
+        let end_offset = make_ceil_byte_offset(self.data_type, end)?;
```
Comment (changhiskhan) [inline rust/src/encodings/plain.rs:189]: does integer division always round down? do we need a `make_floor_byte_offset` for start_offset?

File: rust/src/encodings/plain.rs [hunk 2]
```diff
@@ -242,6 +250,16 @@ impl<'a> PlainDecoder<'a> {
             })?;
         Ok(Arc::new(FixedSizeBinaryArray::try_new(values, stride)?) as ArrayRef)
     }
+
+    async fn take_boolean(&self, indices: &UInt32Array) -> Result<ArrayRef> {
+        // TODO: optimize boolean access
```
Comment (changhiskhan) [inline rust/src/encodings/plain.rs:255]: make a ticket for this?

Comment (eddyxu) [reply rust/src/encodings/plain.rs:255]: Filed https://github.com/eto-ai/lance/issues/629

File: rust/src/encodings/plain.rs (hunk 1 above)
Comment (eddyxu) [reply rust/src/encodings/plain.rs]: yes it should be, given that offset are unsigned int?

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**